include/seastar/core/condition-variable.hh

```diff
@@ -58,6 +59,14 @@ public:
     virtual const char* what() const noexcept;
 };
 
+/// Exception thrown when wait() operation is aborted
+/// \ref condition_variable::wait(abort_source&).
+class condition_variable_aborted : public std::exception {
```

Comment:
Lets inherit from `abort_requested_exception` like we already do for some other abort exceptions.